test/hotspot/jtreg/compiler/lib/ir_framework/test/ArgumentsProvider.java

```diff
@@ -57,36 +50,34 @@ public boolean isDefault() {
      */
     public abstract Object[] getArguments(Object invocationTarget, int invocationCounter);
 
-    /**
-     * For a test method, determine what ArgumentsProvider is to be constructed, given its @Arguments annotation,
-     * and the available setup methods.
-     */
-    public static ArgumentsProvider getArgumentsProvider(Method method,
-                                                         HashMap<String, Method> setupMethodMap) {
+}
+
+/**
+ * For a test method, determine what ArgumentsProvider is to be constructed, given its @Arguments annotation,
+ * and the available setup methods.
+ */
+class ArgumentsProviderBuilder {
+   public static ArgumentsProvider build(Method method,
+                                          HashMap<String, Method> setupMethodMap) {
         Arguments argumentsAnnotation = method.getAnnotation(Arguments.class);
         if (argumentsAnnotation == null) {
             return new DefaultArgumentsProvider();
         }
 
         Argument[] values = argumentsAnnotation.values();
-        String[] setup = argumentsAnnotation.setup();
+        String setupMethodName = argumentsAnnotation.setup();
 
-        if (setup.length > 0) {
-            TestFormat.check(setup.length == 1,
-                             "@Arguments: setup should specify exactly one setup method " +
-                             " but got " + setup.length + " in " + method);
+        if (setupMethodName.length() > 0) {
```

Comment:
I suggest to use `isEmpty()` instead:
```suggestion
        if (!setupMethodName.isEmpty()) {
```